utils/dataset_manifest/core.py

```diff
@@ -355,53 +423,65 @@ def get_subset(self, subset_names):
 class VideoManifestManager(_ManifestManager):
     _requared_item_attributes = {'number', 'pts'}
 
-    def __init__(self, manifest_path):
-        super().__init__(manifest_path)
+    def __init__(self, manifest_path, create_index=True):
+        super().__init__(manifest_path, create_index)
         setattr(self._manifest, 'TYPE', 'video')
         self.BASE_INFORMATION['properties'] = 3
 
-    def create(self, content, **kwargs):
-        """ Creating and saving a manifest file """
-        with open(self._manifest.path, 'w') as manifest_file:
-            base_info = {
-                'version': self._manifest.VERSION,
-                'type': self._manifest.TYPE,
-                'properties': {
-                    'name': os.path.basename(content.source_path),
-                    'resolution': content.frame_sizes,
-                    'length': content.get_size(),
-                },
-            }
-            for key, value in base_info.items():
-                json_item = json.dumps({key: value}, separators=(',', ':'))
-                manifest_file.write(f'{json_item}\n')
-
-            for item in content:
+    def link(self, media_file, upload_dir=None, chunk_size=36, force=False, only_key_frames=False, **kwargs):
+        ReaderClass = VideoStreamReader if not only_key_frames else KeyFramesVideoStreamReader
+        self._reader = ReaderClass(
+            os.path.join(upload_dir, media_file) if upload_dir else media_file,
+            chunk_size,
+            force)
+
+    def _write_base_information(self, file):
+        base_info = {
+            'version': self._manifest.VERSION,
+            'type': self._manifest.TYPE,
+            'properties': {
+                'name': os.path.basename(self._reader.source_path),
+                'resolution': self._reader.resolution,
+                'length': len(self._reader),
+            },
+        }
+        for key, value in base_info.items():
+            json_item = json.dumps({key: value}, separators=(',', ':'))
+            file.write(f'{json_item}\n')
+
+    def _write_core_part(self, file):
+        for item in tqdm(self._reader,
```

Comment:
It is better don't use tqdm inside a library. It is a good solution for create.py. What do you think?

Reply:
I think that need to use `tqdm` where an iteration directly takes place. It seems to me not a very good solution to raise an iteration to a higher level. 
If we will use `tqdm` in create.py, we should change `manifest.create()` to `for _ in manifest.create(): pass` in all places where `manifest.create()` is used, or expand existing API. What do you think about it?

Reply:
Does it spam CVAT server logs with unnecessary messages?

Reply:
In general I don't advice a solution. I just want to say that the general rule is to use tqdm inside CLI not in a library.

Reply:
You can pass an optional object into the `create` method. Thus CLI will pass tqdm like object. If the code is imported, the object will be None.